src/main/java/org/home/implementation/FileMessageStorageImpl.java

```diff
@@ -1,25 +1,101 @@
 package org.home.implementation;
 
 import org.home.interfaces.MessageStorageInterface;
+import org.home.model.FileMessage;
 import org.home.model.Message;
 import org.home.model.MessageQueue;
 import org.springframework.stereotype.Component;
 
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.Queue;
 
-@Component("secondary")
+//@Component("secondary")
 public class FileMessageStorageImpl implements MessageStorageInterface {
-    public FileMessageStorageImpl() {
+    String QFileName;
+    ObjectOutputStream oo;
+    ObjectInputStream io;
+    public FileMessageStorageImpl(String qname) {
+        QFileName = "Queue-"+qname+".txt";
     }
 
     @Override
     public void storeMessage(Message message) {
-
+        try {
+            oo = new ObjectOutputStream(new FileOutputStream(QFileName,true));
+            oo.writeObject(new FileMessage(message.getKey(), message.getBody()));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        finally{
+            if(oo!=null)
+            {
+                try {
+                    oo.close();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
     }
 
     @Override
     public Message fetchMessage() {
+        boolean message_read = false;
```

Comment:
Fix it. 

Every time a read-request is coming, you are opening a File connection, this is extremely inefficient. 
Rather keep it opened when you are constructing this class and re-use the connection whenever you need it.